src/bun.js/api/server.zig

```diff
@@ -1674,7 +1676,16 @@ pub const ServerConfig = struct {
 
             if (try arg.getTruthy(global, "fetch")) |onRequest_| {
                 if (!onRequest_.isCallable()) {
-                    return global.throwInvalidArguments("Expected fetch() to be a function", .{});
+                    if (onRequest_.isUndefined()) {
+                        return global.throwInvalidArguments(
+                            "Bun.serve() requires a fetch() function to handle incoming requests. For more information, see " ++ bun_docs_url ++ "api/http#bun-serve",
```

Comment:
i thought that we should accept routes and/or fetch, so this error message isn't exactly right.

also getTruthy avoids `undefined` already so this is not going to be hit